---
title: "Calculate Nuclear Fraction"
author: "Kennedi Todd"
date: "02/08/2023"
output:
  html_document:
    theme: cerulean
    toc: true
    toc_float: true
editor_options: 
  chunk_output_type: inline
---

# Setup
## Set working directory
```{r working_directory}
knitr::opts_knit$set(root.dir = ".")
```

## Load libraries
```{r load_libraries}
library(DropletQC) # nuclear_fraction_tag()
```

# Calculate intronic read ratio
## snWT
- nuclear fraction = # intronic reads / (# intronic reads + # of exonic reads) \
- The nuclear_fraction_tag() function uses Cell Ranger's output BAM file to calculate nuclear fraction. The BAM file contains tags for reads algined to an intron or exon. \
- This took ~11 minutes with 10 cores. \
```{r wt_nuclear_fraction, eval=FALSE}
# calculate nuclear fraction using BAM tags
dir <- "../../../counts/x10_sn_WT/outs/"
wt.tags <- nuclear_fraction_tags(outs = dir,
                                 cores = 15,
                                 verbose = TRUE)
write.table(wt.tags, "../../../refs/x10_sn_wt_nuclear_fraction.tsv",
            quote = FALSE, sep = "\t", col.names = TRUE, row.names = TRUE)
```

## snAPP
```{r app_nuclear_fraction, eval=FALSE}
# calculate nuclear fraction using BAM tags
dir <- "../../../counts/x10_sn_APP/outs/"
app.tags <- nuclear_fraction_tags(outs = dir,
                                 cores = 15,
                                 verbose = TRUE)
write.table(app.tags, "../../../refs/x10_sn_app_nuclear_fraction.tsv",
            quote = FALSE, sep = "\t", col.names = TRUE, row.names = TRUE)
```
